### PR TITLE
Asteroid Shuttle is Now Free, but Only Purchasable After a Meteor Wave

### PR DESCRIPTION
**Pull request by wesoda25**
## About The Pull Request

Title

## Why It's Good For The Game

I always thought the solution of just bumping up the price was lame, not to mention made 0 sense ICly (hollowing process is pricey when it takes 5 shots from plasma cutter?????). This still will decrease the amount of times its purchased (if anything its harder to get now) but now its more gimmicky, and makes a lot more sense (centcomm harvesting a nearby asteroid and throwing some engines on, so its free). 

## Changelog
:cl:
tweak: Asteroid With Engines Strapped to It shuttle can now be purchased for free, only after a meteor wave
/:cl:

### CODE REVIEW

**Comment by loserXD** [comment]
it's asteroid not meteor

**Comment by SouDescolado** [comment]
Inb4 renamed to meteor shuttle

**Comment by wesoda25** [comment]
i havent a clue what you are talking about

**Comment by FantasticFwoosh** [comment]
Shouldn't this notify after a meteor storm has stopped that it is availible to purchase?

**Comment by wesoda25** [comment]
Theres no notification about arena (AFAIK), I don't really see the point in this. I'll add a little something to the shuttle description though.

**Comment by MrDoomBringer** [comment]
@wesoda25 I think a notification that arena is available would be great, honestly.

**Comment by 81Denton** [comment]
I agree, but it doesn't have to be a global CentCom announcement. A paper printout at comms consoles (like for the roundstart command report) or something similar would be fine too imo.

**Comment by PKPenguin321** [comment]
maybe an automatic PDA message to the heads of staff?

**Comment by Tlaltecuhtli** [comment]
telepatic message to greytiders with the spare

**Comment by dr-bee** [comment]
This mitigates the issue of the asteroid destroying several departments  as if it is after a meteor event those departments would be destroyed anyway, which was the biggest problem with the shuttle.

**Comment by wesoda25** [comment]
> This mitigates the issue of the asteroid destroying several departments as if it is after a meteor event those departments would be destroyed anyway, which was the biggest problem with the shuttle.

Yeah this is a good point I never even thought about, if anything it makes meteor shuttle ideal since crew won't have to walk through breached areas to reach the shuttle. 

Also i will make a centcomm printout for shuttle prereqs i just cant find non-priority announcement code so gimme a sec

**Comment by ExcessiveUseOfCobblestone** [comment]
The implication is that RNG really hecked you over which is not always the case, especially since this applies to all meteor wave types.

I would prefer the cost be kept or lowered but not removed, at the end of the day it's still a grief shuttle.

File: code/modules/events/meteor_wave.dm [hunk 1]
```diff
@@ -52,6 +52,7 @@
 /datum/round_event/meteor_wave/tick()
 	if(ISMULTIPLE(activeFor, 3))
 		spawn_meteors(5, wave_type) //meteor list types defined in gamemode/meteor/meteors.dm
+		SSshuttle.shuttle_purchase_requirements_met |= "meteor"
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/events/meteor_wave.dm]: why is this here?

**Comment by ATH1909** [comment]
Can you buy the asteroid shuttle after a meat-eor wave?

**Comment by ExcessiveUseOfCobblestone** [comment]
yes

**Comment by ExcessiveUseOfCobblestone** [comment]
this will also unlock during the dust storm variant of meteors.

File: code/modules/events/meteor_wave.dm (hunk 1 above)
Comment (wesoda25) [reply code/modules/events/meteor_wave.dm]: umm isn’t this just when meteor waves are spawned?

**Comment by wesoda25** [comment]
Dust storm isn’t intentional so I will blacklist that in a bit. 

Also no, I’m not making it cost money. A huge motivation for this PR was how fucking shitty the PR to make it cost was, it was a horribly disguised jerk reaction band-aid PR. The intention of that PR was to discourage use, this PR does the same except its on RNG as opposed to scrounging up 10K credits.

**Comment by CRITAWAKETS** [comment]
Why not make it so if its after meaty ores the shuttle has meat inside of it?

File: code/datums/shuttles.dm [hunk 2]
```diff
@@ -216,11 +216,16 @@
 /datum/map_template/shuttle/emergency/meteor
 	suffix = "meteor"
 	name = "Asteroid With Engines Strapped To It"
-	description = "A hollowed out asteroid with engines strapped to it, the hollowing procedure makes it very difficult to hijack but is very expensive. Due to its size and difficulty in steering it, this shuttle may damage the docking area."
+	description = "A hollowed out asteroid with engines strapped to it. Due to its size and difficulty in steering it, this shuttle may damage the docking area. Only available after a meteor wave, for both financial and practicality reasons."
```
Comment (bgobandit) [inline code/datums/shuttles.dm]: can you come up with some better lore for this, even what you said in the PR "CentComm does not believe in letting a good meteor wave go to waste."

File: code/modules/events/meteor_wave.dm (hunk 1 above)
Comment (wesoda25) [reply code/modules/events/meteor_wave.dm]: I mean like it seems work, tried to buy it on private server roundstart and it didn’t work, but after triggering meteor event it worked sooo? Is it just shit code that finds a way to work?

**Comment by wesoda25** [comment]
> Why not make it so if its after meaty ores the shuttle has meat inside of it?

This is a funny idea and I’d like to do it but I don’t know mapping, so for the now I can’t do this. Perhaps I can do a follow up later once I learn how?

File: code/modules/events/meteor_wave.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/events/meteor_wave.dm]: why would you have it do this every tick instead of just running it on the start of meteor wave?

File: code/datums/shuttles.dm (hunk 2 above)
Comment (wesoda25) [reply code/datums/shuttles.dm]: wadasdasd

File: code/modules/events/meteor_wave.dm (hunk 1 above)
Comment (wesoda25) [reply code/modules/events/meteor_wave.dm]: bc im code illiterate

Comment (wesoda25) [reply code/modules/events/meteor_wave.dm]: aight so I did this:
if("normal")
  .     wave_type = GLOB.meteors_normal
  .     SSshuttle.shuttle_purchase_requirements_met |= "meteor"
   .     minor_announce("'Asteroid With Engines Strapped To It' Escape Shuttle now available!")

And only did that for normal through catastrophic, excluding all special types of meteor waves. But I need a delay (bc it automatically tells everyone about the shuttle, before an admin announcement (another issue being if admins choose to keep the meteors silent, the new shuttle announcement still gives it away). On top of this, I want it to only make the announcement once, since everytime a meteor wave comes it repeats the shuttle announcement, ideas? I also plan to make an announcement for bubblegum dying, that at least will be more straightforward.

**Comment by ExcessiveUseOfCobblestone** [comment]
> A huge motivation for this PR was how fucking shitty the PR to make it cost was, it was a horribly disguised knee-jerk reaction band-aid PR. The intention of that PR was to discourage use, this PR does the same except its on RNG as opposed to scrounging up 10K credits.

Pretty sure since the beginning of their inception people (at least me) have said having it give you money was cruddy. Calling it knee jerk is just buzzword spaghetti testing. You just didn't like the PR.

Even with the random event it's still:
1) an easy way to kill unsuspecting players because lol???
2) "The shuttle is hijack proof, bomb proof, and is always pressurized. Super easy to survive on it"

having it free because meteors swung around the outskirts of the station seems pretty subpar. My compromise would be to have it scale downwards as threat level rises with it only being free after multiple meteor events have occurred.

**Comment by wesoda25** [comment]
Except its not going to crush anyone since the meteors have destroyed or at least breached evac at this point, xeno for sure will be evacuated. Above all it makes 0 sense ICly that a fucking rock costs 10K credits, I will put it at 2K credits at the highest and possibly scale it down from there.

File: code/modules/events/meteor_wave.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/events/meteor_wave.dm]: just make a list tbh

```DM
    var/static/list/string2meteortype = list(
        "normal" = GLOB.meteors_normal, "threatening" = GLOB.meteors_threatening //etcetc)
    If(!wave_name)
        //don't edit old code it's fine
    wave_type = string2meteortype[wave_type] //die switch die
```
Then in announce
```DM
    if(wave_type in list("normal","threatening","catastrophic") && !("meteor" in SSshuttle.shuttle_purchase_requirements_met))
        SSshuttle.shuttle_purchase_requirements_met |= "meteor"
        minor_announce("'Asteroid With Engines Strapped To It' Escape Shuttle now available!")

**Comment by ExcessiveUseOfCobblestone** [comment]
You don't know that and it's quite possible for a meteor event to hit nothing at all albeit it's definitely related to the map's orientation and size.

I've NEVER claimed there was a lore reason for setting it to a high cost. If we're going by that metric we should nerf Plasma Cutters and mining equipment in general instead because rocks are supposedly really hard to cut and require loads of monopoly money as seen with jerry rigged asteroid+engine shuttle. It goes both ways which is why lore/pseudo-realism is not a (very good) argument. The reason has always been it exists just to grief, although it's picked up a second rationale of being really strong compared to the other shuttles in practicality.

I'm thinking 10K base and 0 minimum (no give monies) > 2.5K drop for Normal, 5K for threat, and 7.5K for Cata. Mix and match to get for free. If you need to satiate your loretism we can just say it takes a very powerful engine to push the thing but having smaller meteors in the vicinity crashing into it propels it for free(tm) and saves space-gas.

**Comment by wesoda25** [comment]
mr cobby i grudgingly accept but i want you to know this upsets me!!

**Comment by Tlaltecuhtli** [comment]
why does the garbage shuttle need to be that expensive? if it is because “muh bomb proof hjack proof” just disable the air generating turfs and add a bunch of goliaths and gibonite ore

**Comment by anconfuzedrock** [comment]
>captain is still paid to get a shitty scrapheap shuttle that screws the crew with  0 upside
>can literally buy a build your own shuttle which has no air
>not fair to buy a meteor shuttle for cheap because it'd have a risk of killing people who don't notice the purchase or even running when they have 10 seconds to before the shuttle lands and escape is likely to be unsafe after meteor impact anyways

cobby is incorrect tbh

**Comment by ExcessiveUseOfCobblestone** [comment]
Any grief shuttle should have a cost associated if not combined with an unlock metric. In terms of cost, the options are "Get rewarded to Grief", "Get to Grief for free" or "You must make an investment to grief"; to me that's a no-brainer.

BYOS is getting made to be less grieffy I think, otherwise yes I think it should cost something too.

No other feature lets you casually kill people regardless of status or justification, so it should be difficult to get the shuttles that permit this. 

>escape is likely ruined

Unless you can show me on the code where it directs meteors to escape I'm not buying the assumption that meteors running = escape is ruined. It depends on map size / layout / severity of meteors and I've already encouraged OP to account for the third option in the case to more precisely account for it rather than shotgun it. I've seen plenty of meteor waves not do anything at all.

> why does the garbage shuttle need to be that expensive? if it is because “muh bomb proof hjack proof” just disable the air generating turfs and add a bunch of goliaths and gibonite ore

Why would you ask a question you clearly know the answer to? If you fix 50% of the issue of course i'm going to be okay with a price drop. I'd like to think i'm fairly reasonable.

**Comment by wesoda25** [comment]
Why the fuck would BYOS shuttle cost money, its a fun project to work on, stop interpreting as any shuttle you dislike as griefy. 

Honestly I fucking hate the attitude that people selectively order this shuttle to just "hee hee i kill people in departures" like holy shit. You guys use the argument that its a shit shuttle and people order it to kill people in one sentence, and that its a God tier, unhijackable shuttle in the next. Make up your fucking minds and stick with one line of reasoning, because the way I see it, the two aren't compatible. And, fun fact, it IS hijackable (wtF EPCI?!!). 

Yeah, I'll put some sort of cost if you won't merge me otherwise. But I think its fucking stupid, and all it does is make it even harder to get asteroid shuttle (bc now its RNG + paying), which goes against the reason I made it. I want it to be niche, I want it to be cool, but no one is gonna get it even when it becomes available now. And thats really fucking lame.

**Comment by ExcessiveUseOfCobblestone** [comment]
> a shuttle that enables killing people for no reason other than mega luls is not grief, you're just projecting your dislike

looks like we'll have to agree to disagree then.

To clarify, the asteroid shuttle is a "grief shuttle" on entry since it gibs if you're not paying attention because it lands on a part of the station. It is a perfectly viable shuttle after that, arguably better than most of the selection. I don't consider that to be conflicting thoughts.

I was copypasting shadow's post from his asteroid cost PR, but the shuttle is bomb proof and much more difficult to hijack given that it's not lit and pretty massive compared to the other guys. If you address that directly, again, I'll be happy to consider lowering the cost. Until then you'll have to suffice with my initial compromise of lowering based on which event(s) is/are ran.

**Comment by ExcessiveUseOfCobblestone** [comment]
I don't plan on BYOS having a cost because the bad aspects are being fixed in a PR.

If that wasn't an option though I would definitely entertain the idea of it coming at a price (well I'd probably just fix it myself, giving a cost to it seems counter-intuitive to the purpose of the shuttle).

File: code/modules/events/meteor_wave.dm (hunk 1 above)
Comment (wesoda25) [reply code/modules/events/meteor_wave.dm]: Second part is good, I added the !meteor and it fixed the repeating. How would I add a delay to the announcement? 

Also do you have any clue my my PR failed the check?

**Comment by knacker48** [comment]
Just out of curiosity, how common is the meteor event?

**Comment by anconfuzedrock** [comment]
Basically my argument cobby is “people who die to this kinda deserve it” so ignore all I say cobbs if that sounds cunty to you

File: code/modules/events/meteor_wave.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/events/meteor_wave.dm]: the announcement should only run when the other announcement is ran hence why I said to put it in `announce()`

**Comment by ExcessiveUseOfCobblestone** [comment]
> Basically my argument cobby is “people who die to this kinda deserve it” so ignore all I say cobbs if that sounds cunty to you

I'd question why them deserving to die makes it less of a grief shuttle/not a grief shuttle at all, when the metric should be the fact it is intended to kill people (even if it's just dumb ones).

File: code/modules/events/meteor_wave.dm (hunk 1 above)
Comment (wesoda25) [reply code/modules/events/meteor_wave.dm]: The other announce is just a fake announcement tho, should i create another announce proc thingy for it separately?

**Comment by anconfuzedrock** [comment]
I don't think of the shuttle as intended to kill people, is my thing. the scrapheap is intended to get people fighting and spacing each other, there is no advantage to purchasing it. This shuttle has some interesting and redeeming qualities, such as being enterable from other parts of the station because of how thicc it is, being thicc so bombs aren't as good, and often being so fat that it crashes into a teleport beacon and people can use that to get on after it leaves. People dying to it is a very preventable side effect that I don't think comes as a concern too often, and I don't think this shuttle is the best choice of you just want to get people killed. we have a lot of cramped shuttles for that already and the cancer bubblegum one.

**Comment by ExcessiveUseOfCobblestone** [comment]
Right, I said I considered it a grief shuttle on entry but after it becomes a really viable shuttle which is even more reason to make it come at a price.

I guess we'll just have to agree to disagree on the dying aspect. I don't consider the ease of not dying a metric, rather that the opportunity to be killed by it if not paying attention is presented. The only comparable one off the top of my head is Oh Hi Daniel and it can only be harmful is someone does something while on the shuttle. Arena is very close to how I want this balanced.

**Comment by wesoda25** [comment]
Arena isn’t even balanced at all tho. 10K and you everyone to kill themselves.

**Comment by ExcessiveUseOfCobblestone** [comment]
If arena was bought as much as the asteroid I would agree. I still do actually, but I'm fine with the rarity of it since people seem to enjoy it (much like the asteroid).

**Comment by SpaceManiac** [comment]
travis failing
```
code/datums/shuttles.dm:	description = "A hollowed out asteroid with engines strapped to it. Due to its size and difficulty in steering it, this shuttle may damage the docking area. Only available after a meteor wave, Centcomm won't let even an asteroid go to waste."
Misspelling(s) of CENTCOM detected in code, please remove the extra M(s).
```

**Comment by wesoda25** [comment]
I’ll just do nanotrasen.

**Comment by wesoda25** [comment]
yeah im not gonna bother finishing this PR

ill cap it off later when the inspiration returns to me

**Comment by wesoda25** [comment]
shit man im trying to resurrect this but keep running into errors
